### DISCUSSION TITLE
repeated replicates across each time step

### DISCUSSION THREAD

**Discussion by Sylvanben** [davidbolin/MetricGraph#15] [Q&A]
Hi,

Thank you for the help in learning the package so far.

I am building a river temperature model which has daily repeated measurements across ~50 sites for 2 years. I want to retain all data points but apply a temporal time step at a much lower resolution (seasonal which has ~90 repeat points each step). I had thought I could add the points on a daily time step using **group**

_graph$add_observations(data = covs_full, data_coords = "spatial", 
                       coord_x = "Xkm", coord_y = "Ykm",
                       #daily repeat points --
                       group=c("date_f"))_

and apply a **repl** column of season 

_st_spde<-graph_spde(graph)

st_data <- graph_data_spde(st_spde,  loc_name = "loc", repl = ".all",
                           repl_col = "season_n")_

However, it looks like these arguments need to be the same. and running the model with replicates for every day is too large to run. This is following your inlabru implementation with replicates example.

I have also tried your spatio-temporal inlabru model example:

_mesh_time<-fmesher::fm_mesh_1d(as.numeric(covs_full$season_f))

st_spde<-rspde.spacetime(mesh_space = graph,
                mesh_time = mesh_time,
                alpha = 1, beta = 1)

st_data <- graph_data_rspde(st_spde, name = "loc", bru=T, repl = ".all",
                            repl_col = "date_f")

##spacetime formula ----------
cmp_graph <- waterTempMax ~ 1 + field(list(
  space = cbind(.edge_number, .distance_on_edge), 
  time = time), 
  model = st_spde,
  replicate = st_data[["repl"]])

bru_fit <-
  bru(cmp_graph, data=st_data[["data"]],
      options=list(verbose=T)
  )_

However, I get the following error:

_*** ERROR *** 	
 *** dlsym error with model[inla_cgeneric_rspde_spacetime_model] err_msg[The specified procedure could not be found.]

Error: C stack usage  156925197 is too close to the limit_

Any help in progressing this would be much appreciated,
Sylvan

**Comment by davidbolin** [comment]
Hi Sylvan, 

To use replicate for a spacetime model you need to use the advanced grouping feature. You can see an example here:

https://davidbolin.github.io/MetricGraph/articles/spacetime.html#fitting-model-with-replicates-in-inlabru

and the grouping feature are explained in more detail here:

https://davidbolin.github.io/MetricGraph/articles/metric_graph_data.html#more-advanced-grouping

The inla error indicates that the inla install does not have the binary for metricgraph. Have you installed the latest development version of inla?

Best,
David

**Reply by Sylvanben** [reply]
Hi David,

Thanks for this advice, I am only getting back to it now after the Christmas break. The cgeneric error has gone but I am still getting an c stack error, with the last few lines of my verbose being: 

file=[C:/Users/SBENAK~1/AppData/Local/Temp/RtmpE30IDk/file29804a367b0b/data.files/file29802febb7c] 7/1851319  (idx,y) = (7, 7)
file=[C:/Users/SBENAK~1/AppData/Local/Temp/RtmpE30IDk/file29804a367b0b/data.files/file29802febb7c] 8/1851319  (idx,y) = (8, 8)
file=[C:/Users/Assertion failed: !strcasecmp(data->ints[9]->name, "bounded_rho"), file cgeneric_rspde_spacetime.c, line 42
Error: C stack usage  1436917216 is too close to the limit

online it says this could be a memory issue but I think bru should be well able to handle the reduced model I have fit, and cgeneric is mentioned in the last line of the verbose so this may still be the issue. I have updated to the most recent testing version of INLA and INLAbru.

Many thanks,
Sylvan

**Comment by davidbolin** [comment]
Hi Sylvan, hard to tell what is wrong without seeing the code, can you link some reproducible example here or email it to me?
